src/v/storage/disk_log_impl.cc

```diff
@@ -1908,6 +1908,19 @@ disk_log_impl::disk_usage_and_reclaim(gc_config cfg) {
     co_return std::make_pair(usage, reclaim);
 }
 
+/*
+ * assumes that the compaction gate is held.
+ */
```

Comment:
Oh that'd be cool. Unfortunately I don't think there is: short of passing around parameters, we don't have fiber-local storage to track that sort of thing. the comment is probably sufficient here since these methods are all private to the implementation.